p2p/chain.py

```diff
@@ -218,10 +218,29 @@ def register_peer(self, peer: BasePeer) -> None:
             # TODO: Process headers for consistency.
             try:
                 head_number = await self._process_headers(peer, headers)
+            except HeaderNotFound:
+                # This is comes from when we attempt to compute the difficulty
+                # of a header whos parent is not in our database.
+                if start_at == 0:
+                    self.logger.warn("No common ancestry found from %s, aborting sync", peer)
+                    break
+                start_at = max(0, start_at - eth.MAX_HEADERS_FETCH)
+                if start_at < head.block_number - 10000:
+                    self.logger.warn(
+                        "No common ancestor found with %s within 10,000 blocks, "
```

Comment:
Nitpick: number should better come from variable (at least, if not even outside constant)